index.ts

```diff
@@ -4,38 +4,57 @@ import { spawn } from 'child_process';
 
 setGracefulCleanup();
 
-type ProjectMutator = (project: Project) => void | Promise<void>;
+/**
+ * A callback used by some of classes to obtain a {@link Project}.
+ * 
+ * You are supposed to pass such a callback. In it, you should create and return a {@link Project}
+ * instance.
+ * 
+ * @returns A {@link Project} instance, optionally wrapped into a promise.
+ */
+
```

Comment:
```suggestion
```

shouldn't this comment block be "connected" to the following code for it to be picked up? or is it smart enough to know that it refers to the next line of code that isn't whitespace?